core/validation/failures.py

```diff
@@ -465,6 +465,21 @@ def to_message(self) -> tuple[str | None, str | None, str]:
         return None, None, message
 
 
+@dataclass
+class NullInputPreTransformationFailure(PreTransFormationFailure):
+    """Class representing a pre-transformation failure where the input is null."""
+
+    tab: str
+    section: str
+    message: str
+
+    def __str__(self):
+        pass
+
+    def to_message(self) -> tuple[str, str, str]:
+        return self.tab, self.section, self.message
```

Comment:
I think we can renamed this to `GenericPreTransformationFailure` or similar as it's not specific to that failure in anyway other than by name

Reply:
Reflecting on it, PreTransFormationFailures don't make use of the tab or section, but for a sign-off failure we need both for the validate message to be more specific to the user. We need to collect information about the tab, section, missing_value, and sign_off_officer, and pass these to the message string, which probably means this particular failure isn't reusable for other generic pre-transformation failures.

I've renamed the failure SignOffFailure, and refactored it such that it will utilise the tab and section but still be treated as a PreTransFormationFailure.